pkgs/by-name/og/ogen/package.nix

```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
+  pname = "ogen";
+  version = "1.3.0";
+
+  src = fetchFromGitHub {
+    owner = "ogen-go";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-aVUCGAVIRv0DFpoIGXyRYBZyQ2URnA0slv9WoKtYK1w=";
+  };
+
+  vendorHash = "sha256-vT67stjOwMwfcYZGcvsk+P0VcunKPaBp8tgZDYufgto=";
+
+  patches = [ ./ogen.patch ];
```

Comment:
We often override the version string (somewhere, haven't found it), if it's not a constant. example: https://github.com/NixOS/nixpkgs/blob/574d1eac1c200690e27b8eb4e24887f8df7ac27c/pkgs/applications/audio/go-musicfox/default.nix#L31

If we can't, use `substituteInPlace`. Updating patch manually isn't expected. Also patch should have a file name that clearly indicates its role

Reply:
Thanks for the feedback @Aleksanaa - I'm aware of this version string mechanism and have used it before; it doesn't work in this case because the program is designed to pull the versioning stuff from git directly using the go `debug/BuildInfo` mechanism (https://pkg.go.dev/runtime/debug#BuildInfo).

I initially tried to use `substituteInPlace` but I could not find a solution which I thought reasonable - as you can see from the patch, I wanted to replace the string `return "", false` with the version number but this is quite a common string and I thought it more robust to make this change via a patch where line numbers are specified where this could/should happen and also ensure that the substitution only happens once.

I've kept it as a patch for now but happy to discuss further etc.

Reply:
Another option could prob be to not use `fetchFromGithub` and to use a fetcher which performs a git clone - I know this is perceived as a more expensive operation and hence is generally advised against, but perhaps it's actually the right solution in this case. That would prob obviate the need for the patch and make this simpler - any thoughts @Aleksanaa?